packages/framework/src/Model/Category/CategoryFacade.php

```diff
@@ -435,4 +438,27 @@ public function getListableProductCountsIndexedByCategoryId(
             $domainId
         );
     }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Category\Category $category
+     * @param array $originalNames
+     */
+    protected function createFriendlyUrlWhenRenamed(Category $category, array $originalNames): void
```

Comment:
This code is too complex, better would be to create only array of changed names and createUrls same as it was done in previous version on line 155.

Reply:
In previous version frienly urls were creaated for all domains every time.

Anyway, I reduced complexity in https://github.com/shopsys/shopsys/pull/1972/commits/cada8e61ed46f9ba8764d5d84d674a005564175b